ui/components/ui/survey-toast/survey-toast.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState, useContext, useMemo } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchWithCache from '../../../../shared/lib/fetch-with-cache';
+import { DAY } from '../../../../shared/constants/time';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import {
+  getSelectedInternalAccount,
+  getLastViewedUserSurvey,
+  getUseExternalServices,
+  getMetaMetricsId,
+  getParticipateInMetaMetrics,
+} from '../../../selectors';
+import { ACCOUNTS_API_BASE_URL } from '../../../../shared/constants/accounts';
+import { setLastViewedUserSurvey } from '../../../store/actions';
+import { Toast } from '../../multichain';
+
+type Survey = {
+  url: string;
+  description: string;
+  cta: string;
+  id: number;
+};
+
+export function SurveyToast() {
+  const [survey, setSurvey] = useState<Survey | null>(null);
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const lastViewedUserSurvey = useSelector(getLastViewedUserSurvey);
+  const participateInMetaMetrics = useSelector(getParticipateInMetaMetrics);
+  const basicFunctionality = useSelector(getUseExternalServices);
+  const internalAccount = useSelector(getSelectedInternalAccount);
+  const metaMetricsId = useSelector(getMetaMetricsId);
+
+  const surveyUrl = useMemo(
+    () => `${ACCOUNTS_API_BASE_URL}/v1/users/${metaMetricsId}/surveys`,
+    [metaMetricsId],
+  );
+
+  useEffect(() => {
+    if (!basicFunctionality || !metaMetricsId || !participateInMetaMetrics) {
+      return undefined;
+    }
+
+    const controller = new AbortController();
+
+    const fetchSurvey = async () => {
```

Comment:
you can useCallback for fetchSurvey
`const fetchSurvey = useCallback(async () => {
...
 }, [
    internalAccount?.address,
    lastViewedUserSurvey,
    basicFunctionality,
    metaMetricsId,
    fetchSurvey,
  ]);
`